src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -609,25 +609,30 @@ public function next_token(): bool {
 		 *       until there are events or until there are no more
 		 *       tokens works in the meantime and isn't obviously wrong.
 		 */
-		while ( empty( $this->element_queue ) && $this->step() ) {
-			continue;
+		if ( empty( $this->element_queue ) && $this->step() ) {
+			return $this->next_token();
 		}
 
 		// Process the next event on the queue.
 		$this->current_element = array_shift( $this->element_queue );
 		if ( ! isset( $this->current_element ) ) {
-			return false;
+			// There are no tokens left, so close all remaining open elements.
+			while ( $this->state->stack_of_open_elements->pop() ) {
+				continue;
+			}
```

Comment:
Minor, but something I've wondered about. the `continue` doesn't do anything here. Is there a reason to include it?
```suggestion
			while ( $this->state->stack_of_open_elements->pop() ) {}
```

Reply:
I think you would have to put a code comment inside at minimum to document the intent otherwise.

Reply:
originally I had a comment inside these loops but WPCS thought it was bad so required the introduction of `continue`, which surprisingly, introduces actual executable code and slows down the runtime

Reply:
How strict WPCS is for `while ( $this->state->stack_of_open_elements->pop() ) {}` in a single line?

Reply:
hmm. I might not have tried that, but I'm personally not a fan because of unexplained empty body. we can prep a PR and see if it complains. do you consider that better?

Reply:
It was only thought exercise as it might go against a linting rule if the new line after an opening curly bracket is enforced.

Reply:
All good with the code that landed 👍